xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -1919,7 +1919,8 @@ void CVideoPlayer::HandlePlaySpeed()
       m_VideoPlayerAudio->SendMessage(new CDVDMsgDouble(CDVDMsg::GENERAL_RESYNC, m_clock.GetClock()), 1);
     }
     else if (m_CurrentVideo.syncState == IDVDStreamPlayer::SYNC_WAITSYNC &&
-             m_CurrentVideo.avsync == CCurrentStream::AV_SYNC_CONT)
+             (m_CurrentVideo.avsync == CCurrentStream::AV_SYNC_CONT ||
+              m_CurrentAudio.syncState == IDVDStreamPlayer::SYNC_INSYNC))
```

Comment:
@FernetMenta need your opinion here: I play stream with chapters from inputstream.adaptive, each chapter devided with an DMX_SPECIALID_STREAMCHANGE. Syncinc stalls between the chapters (depending the stream length up to ~10 seconds). In this case m_CurrentVideo.avsync is CCurrentStream::AV_SYNC_NONE during stall, means that it will never be set to CONT.
The code change is identical to what is done some lines above for Audio. What do you think?

Reply:
This change is definitely wrong. If your video stream stalls, you are most like doing something wrong setting the stream parameters. OpenDefaultStream / OpenVideoStream won't reopen video stream if config did not change.

Reply:
btw. this is another example where you hide non trivial changes in some PR with a title nobody would expect those kind of changes in.

Reply:
@FernetMenta  I doubt that this error is caused by flags passed from outside.
Even if it would be like that, the combination m_CurrentVideo.syncState == IDVDStreamPlayer::SYNC_WAITSYNC and  m_CurrentVideo.avsync == CCurrentStream::AV_SYNC_NONE should never appear, right?

Reply:
> Even if it would be like that, the combination m_CurrentVideo.syncState == IDVDStreamPlayer::SYNC_WAITSYNC and m_CurrentVideo.avsync == CCurrentStream::AV_SYNC_NONE should never happen, right?

Wrong, this is quite a common state.

You should investigate why your streamchange puts VPV into state SYNC_WAITSYNC. In general that happens when a new video stream is opened. Why does this happen if parameters do not change?

Reply:
Every chapter has its own stream id collection (1000 - 1999 for first chapter, 2000->2999 for second and so on). This means that for every chapter DMX_SPECIALID_STREAMCHANGE retrieves new stream id's. This leads to OpenStream, and VPV::OpenStream sets playerstate to SYNC_START, which will be set do WAIT_SYNC in VP if the first decoder output arrives.

https://paste.ubuntu.com/p/g9Z7RyJbQJ/

Log contains playback of a chapter stream, 2 adverts (streamId 100x and 200x) and then the main movie (300x) issue is at this time:

2019-09-19 16:27:39.660 T:4294967294   DEBUG: CVideoPlayer::HandleMessages - player started 2. After this there is 7 seconds only the first frame displayed.

Reply:
did you record the log with all the changes in this PR? Without knowing the corresponding code the log is useless for me.

Reply:
The log is with kodi master, this PR not included.

Reply:
I would have assumed that m_currentVideo.avsync is first set to AV_SYNC_CHECK, then to AV_SYNC_CONT. Why is this AV_SYNC_NONE in your case?

Reply:
https://paste.ubuntu.com/p/4pBnk53Rm6/ is a log where I print every change of avsync (both video and audio) with line numbers. Line numbers have not changed and should match kodi master. 

```
2019-09-21 19:20:15.746 T:4294967294   DEBUG: XXX 2 CONT 2254
2019-09-21 19:20:15.746 T:4294967294   DEBUG: XXX V NONE 1655
2019-09-21 19:20:15.747 T:4294967294  NOTICE: DXVA::CDecoder::Close: closing decoder.
2019-09-21 19:20:15.747 T:4294967294   DEBUG: DXVA::CVideoBufferPool::~CVideoBufferPool: destructing buffer pool.
2019-09-21 19:20:15.756 T:4294967294   DEBUG: CVideoPlayer::HandleMessages - player started 2
2019-09-21 19:20:22.315 T:4294967294    INFO: XCURL::DllLibCurlGlobal::CheckIdle - Closing session to https://dcs-vod.apis.anvato.net (easy=0x16ecb6e8, multi=0x165fd700)
2019-09-21 19:20:22.585 T:4294967294   DEBUG: VideoPlayer::Sync - Video - pts: 23757000.000000, cache: 50000.000000, totalcache: 100000.000000
```

Reply:
Strange, why is it not set to CONT in CheckContinuity? There is no CheckContinuity log that would indicate discontinuity of the timestamps.

Reply:
Hmm, it is set to CONT. Why evaluates condition in line 1654 to true?

Reply:
the state of avsync should not change that often as it does in your log.

Reply:
Ok, I'll first try to figure out why it changes that often

Reply:
https://paste.ubuntu.com/p/TQVrbD7mHv/
The old log was not ok, sry.

The stall ist now here:
`2019-09-21 21:51:40.834 T:4294967294   DEBUG: CVideoPlayer::HandleMessages - player started 2`

Maybe the issue is caused by the very short length of the stream 2001/2, it is only a few seconds, but right ahead the stream which have the 7sec stall

Reply:
Can VP handle the situation that 3 streams are in "queue" ? we have 1001 playing, 2001 is very short, and we start already with 3001.

Reply:
@FernetMenta I created an example for reproduction, maybe it helps:
https://paste.ubuntu.com/p/JMqb4ptWCJ/, instructions in the paste.